web/src/main/java/org/cbioportal/web/GeneController.java

```diff
@@ -19,37 +30,75 @@
     @ApiOperation("Get all genes")
-    public ResponseEntity<List<Gene>> getAllGenes(@RequestParam(defaultValue = "SUMMARY") Projection projection,
-                                                                   @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_SIZE) Integer pageSize,
-                                                                   @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_NUMBER) Integer pageNumber) {
+    public ResponseEntity<List<Gene>> getAllGenes(
+            @ApiParam("Level of detail of the response")
+            @RequestParam(defaultValue = "SUMMARY") Projection projection,
+            @ApiParam("Page size of the result list")
+            @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_SIZE) Integer pageSize,
+            @ApiParam("Page number of the result list")
+            @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_NUMBER) Integer pageNumber,
+            @ApiParam("Name of the property that the result list is sorted by")
+            @RequestParam(required = false) GeneSortBy sortBy,
+            @ApiParam("Direction of the sort")
+            @RequestParam(defaultValue = "ASC") Direction direction) {
 
-        throw new UnsupportedOperationException();
+        if (projection == Projection.META) {
+            HttpHeaders responseHeaders = new HttpHeaders();
+            responseHeaders.add(HeaderKeyConstants.TOTAL_COUNT, geneService.getMetaGenes().getTotalCount()
+                    .toString());
+            return new ResponseEntity<>(responseHeaders, HttpStatus.OK);
+        } else {
+            return new ResponseEntity<>(
+                    geneService.getAllGenes(projection.name(), pageSize, pageNumber,
+                            sortBy == null ? null : sortBy.name(), direction.name()), HttpStatus.OK);
+        }
     }
 
-    @RequestMapping(value = "/genes/{entrezGeneId}", method = RequestMethod.GET)
+    @RequestMapping(value = "/genes/{geneId}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
     @ApiOperation("Get a gene")
-    public ResponseEntity<Gene> getGene(@PathVariable String entrezGeneId) {
+    public ResponseEntity<Gene> getGene(
+            @ApiParam(required = true, value = "Entrez Gene ID or Hugo Gene Symbol e.g. 1 or A1BG")
+            @PathVariable String geneId) throws GeneNotFoundException {
 
-        throw new UnsupportedOperationException();
+        return new ResponseEntity<>(geneService.getGene(geneId), HttpStatus.OK);
     }
 
-    @RequestMapping(value = "/genes/fetch", method = RequestMethod.POST)
-    @ApiOperation("Fetch genes by ID")
-    public ResponseEntity<List<Gene>> fetchGenes(@RequestParam(defaultValue = "SUMMARY") Projection projection,
-                                                                  @RequestBody List<String> entrezGeneIds) {
+    @RequestMapping(value = "/genes/{geneId}/aliases", method = RequestMethod.GET,
```

Comment:
as mentioned above .. I think we will eventually need a "fetch by alias" function .. either as a separate endpoint or perhaps as additional functionality in the /genes/{geneId} endpoint ... making that work to also do alias resolution. But it is probably better to have a separate "alias-included" query from an "exact match" query